src/data/bool/basic.lean

```diff
@@ -225,4 +225,8 @@ by cases b; simp only [of_nat,to_nat]; exact dec_trivial
 ⟨λ Hinj Heq, ff_ne_tt (Hinj Heq),
   λ H x y hxy, by { cases x; cases y, exacts [rfl, (H hxy).elim, (H hxy.symm).elim, rfl] }⟩
 
+/-- **Kaminski's Equation** -/
+@[simp] theorem apply_apply_apply (f : bool → bool) (x : bool) : f (f (f x)) = f x :=
```

Comment:
per @semorrison's [comment on Zulip](https://leanprover.zulipchat.com/#narrow/stream/217875-Is-there-code-for-X.3F/topic/Kaminski's.20Equation/near/282407174):
```suggestion
theorem apply_apply_apply (f : bool → bool) (x : bool) : f (f (f x)) = f x :=
```